### ISSUE TITLE
Mining factories with players in them

### ISSUE THREAD

**Issue by Adilf** [sirenfal-factorio/Free_Real_Estate#1]
Mining a factory containing online player results in 
```
__Free_Real_Estate__/factory/factory.lua:477: attempt to index global 'p' (a nil value)
3943.397 Error ServerMultiplayerManager.cpp:93: MultiplayerManager failed: "Error while running event on_preplayer_mined_item (ID 11)
__Free_Real_Estate__/factory/factory.lua:477: attempt to index global 'p' (a nil value)"
```
Mining a factory with offline player traps him inside.
Placement of a factory containing a player that has just reconnected results in single desync.

**Comment by avosirenfal** [comment]
I don't have anyone on to test right now but this should fix the crash. I'm not sure it fixes the desync. I'm about to go to bed but I'll look into it in detail when I wake up if you're still having issues.

https://github.com/sirenfal-factorio/Free_Real_Estate/releases/download/1.0.1/Free_Real_Estate_1.0.1.zip

**Comment by avosirenfal** [comment]
This should be fixed in 1.0.1. I didn't test it with recursion because I'm tired, but it seems to work fine without recursion enabled.

Thanks for the report, let me know if you find anything else.

Download: https://github.com/sirenfal-factorio/Free_Real_Estate/releases/download/1.0.1/Free_Real_Estate_1.0.1.zip
